### PR TITLE
UPC-58 New property to tell if a plugin is supported by SonarSource or not

### PR DESCRIPTION
**Pull request by erichirlemann**

### CODE REVIEW

File: sonar-update-center-mojo/src/main/java/org/sonar/updatecenter/mojo/PluginHeader.java [hunk 1]
```diff
@@ -143,6 +143,8 @@ public int getNbVersions() {
     return plugin.getAllReleases().size();
   }
 
+  public boolean isSupported() { return plugin.isSupported(); }
```
Comment (SonarTech) [inline sonar-update-center-mojo/src/main/java/org/sonar/updatecenter/mojo/PluginHeader.java]: MAJOR Make this line start at column 5. [rule](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AIndentationCheck)

**Comment by SonarTech** [comment]
SonarQube analysis reported 1 issue:
* MAJOR 1 major

Watch the comments in this conversation to review them.

File: sonar-update-center-common/src/main/java/org/sonar/updatecenter/common/Plugin.java [hunk 2]
```diff
@@ -36,6 +36,7 @@
   private String category;
   private String issueTrackerUrl;
   private String sourcesUrl;
+  private boolean supported = false;
```
Comment (henryju) [inline sonar-update-center-common/src/main/java/org/sonar/updatecenter/common/Plugin.java]: I would be more explicit here. supportedBySonarSource "supported" could mean supported in a given SQ server version.

**Comment by henryju** [comment]
I would replace any occurence of "supported" by "supportedBySonarSource" to avoid any confusion.

**Comment by SonarTech** [comment]
SonarQube analysis reported 5 issues:
* MAJOR 4 major
* MINOR 1 minor

Watch the comments in this conversation to review them.
Note: the following issues could not be reported as comments because they are located on lines that are not displayed in this pull request:
* MAJOR [Replace the type specification in this constructor call with the diamond operator ("<>").](https://github.com/SonarSource/sonar-update-center/blob/0e377b1b569c4be896c49b7b48bf402e5910858b/sonar-update-center-common/src/main/java/org/sonar/updatecenter/common/UpdateCenterDeserializer.java#L173) [rule](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS2293)
* MAJOR [Make "loadPluginProperties" a "static" method.](https://github.com/SonarSource/sonar-update-center/blob/0e377b1b569c4be896c49b7b48bf402e5910858b/sonar-update-center-common/src/main/java/org/sonar/updatecenter/common/UpdateCenterDeserializer.java#L115) [rule](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS2325)
* MAJOR [Make "resolve" a "static" method.](https://github.com/SonarSource/sonar-update-center/blob/0e377b1b569c4be896c49b7b48bf402e5910858b/sonar-update-center-common/src/main/java/org/sonar/updatecenter/common/UpdateCenterDeserializer.java#L370) [rule](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS2325)
* MAJOR [Use "Character.toString" instead.](https://github.com/SonarSource/sonar-update-center/blob/0e377b1b569c4be896c49b7b48bf402e5910858b/sonar-update-center-common/src/main/java/org/sonar/updatecenter/common/UpdateCenterDeserializer.java#L361) [rule](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS2131)
* MINOR [Define a constant instead of duplicating this literal " in: " 3 times.](https://github.com/SonarSource/sonar-update-center/blob/0e377b1b569c4be896c49b7b48bf402e5910858b/sonar-update-center-mojo/src/main/java/org/sonar/updatecenter/mojo/PluginHeaders.java#L86) [rule](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS1192)

**Comment by SonarTech** [comment]
SonarQube analysis reported 3 issues:
* MAJOR 2 major
* MINOR 1 minor

Watch the comments in this conversation to review them.
Note: the following issues could not be reported as comments because they are located on lines that are not displayed in this pull request:
* MAJOR [Make "resolve" a "static" method.](https://github.com/SonarSource/sonar-update-center/blob/7f37ac3be5ee3d4d7d42da60467352b68eae9a7b/sonar-update-center-common/src/main/java/org/sonar/updatecenter/common/UpdateCenterDeserializer.java#L370) [rule](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS2325)
* MAJOR [Use "Character.toString" instead.](https://github.com/SonarSource/sonar-update-center/blob/7f37ac3be5ee3d4d7d42da60467352b68eae9a7b/sonar-update-center-common/src/main/java/org/sonar/updatecenter/common/UpdateCenterDeserializer.java#L361) [rule](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS2131)
* MINOR [Define a constant instead of duplicating this literal " in: " 3 times.](https://github.com/SonarSource/sonar-update-center/blob/7f37ac3be5ee3d4d7d42da60467352b68eae9a7b/sonar-update-center-mojo/src/main/java/org/sonar/updatecenter/mojo/PluginHeaders.java#L86) [rule](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS1192)

**Pull request merged**